internal/controller/conditions/conditions.go

```diff
@@ -0,0 +1,87 @@
+package conditions
+
+import (
+	"context"
+	"fmt"
+
+	siteconfigv1 "github.com/sakhoury/siteconfig/api/v1alpha1"
+	"github.com/sakhoury/siteconfig/internal/controller/retry"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ConditionType is a string representing the condition's type
+type ConditionType string
+
+// ConditionTypes define the different types of conditions that will be set
+var ConditionTypes = struct {
+	SiteConfigValidated        ConditionType
+	RenderedTemplates          ConditionType
+	RenderedTemplatesValidated ConditionType
+	Provisioning               ConditionType
```

Comment:
What is the distinction between `Provisioning` and `InstallInProgress`?

Reply:
`InstallInProgress` and `InstallCompleted` are remnants of earlier poc work. I have removed them.

Reply:
Minor nit, some of the conditions seem to be thing-state (eg SiteConfigRendered) and others seem to be state-thing (eg RenderedTemplates). For consistency, and clarity of what the state is, perhaps:
```suggestion
	SiteConfigValidated        ConditionType
	TemplatesRendered          ConditionType
	RenderedTemplatesValidated ConditionType
	Provisioned               ConditionType```

Reply:
Changed to `Provisioned`